features/190-consistent-error/api.md

```diff
@@ -0,0 +1,95 @@
+# Consistent error handling
+
+## Requirement
+
+* To support a consistent way of handling error
+* To obtain error information from failed API request
+* To improve TypeScript support with regards to data obtained from rejected
+  promise.
+* Support legacy browsers (aka pre-IE-11).
```

Comment:
Product Design-wise, if it induces too much complexity or difficulties, I think we can drop pre-IE-11 browser support.